bluetooth/src/iosMain/kotlin/device/DeviceConnectionManager.kt

```diff
@@ -0,0 +1,224 @@
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.bluetooth.device
+
+import com.splendo.kaluga.base.toNSData
+import com.splendo.kaluga.base.typedList
+import com.splendo.kaluga.bluetooth.DefaultServiceWrapper
+import com.splendo.kaluga.bluetooth.Service
+import com.splendo.kaluga.bluetooth.uuidString
+import com.splendo.kaluga.logging.info
+import kotlinx.coroutines.launch
+import platform.CoreBluetooth.CBCentralManager
+import platform.CoreBluetooth.CBCharacteristic
+import platform.CoreBluetooth.CBDescriptor
+import platform.CoreBluetooth.CBPeripheral
+import platform.CoreBluetooth.CBPeripheralDelegateProtocol
+import platform.CoreBluetooth.CBService
+import platform.CoreBluetooth.CBUUID
+import platform.Foundation.NSError
+import platform.Foundation.NSNumber
+import platform.darwin.NSObject
+import kotlin.coroutines.CoroutineContext
+
+internal actual class DeviceConnectionManager(
+    private val cbCentralManager: CBCentralManager,
+    private val peripheral: CBPeripheral,
+    connectionSettings: ConnectionSettings,
+    deviceHolder: DeviceHolder,
+    stateRepo: DeviceStateFlowRepo,     
+) : BaseDeviceConnectionManager(connectionSettings, deviceHolder, stateRepo) {
+
+    class Builder(private val cbCentralManager: CBCentralManager, private val peripheral: CBPeripheral) : BaseDeviceConnectionManager.Builder {
+        override fun create(
+            connectionSettings: ConnectionSettings,
+            deviceHolder: DeviceHolder,
+            stateRepo: DeviceStateFlowRepo
+        ): BaseDeviceConnectionManager {
+            return DeviceConnectionManager(
+                cbCentralManager,
+                peripheral,
+                connectionSettings,
+                deviceHolder,
+                stateRepo
+            )
+        }
+    }
+
+    companion object {
+        private const val TAG = "IOS Bluetooth DeviceConnectionManager"
+    }
+
+    private val discoveringServices = emptyList<CBUUID>().toMutableList()
+    private val discoveringCharacteristics = emptyList<CBUUID>().toMutableList()
+
+    @Suppress("CONFLICTING_OVERLOADS")
+    private val peripheralDelegate = object : NSObject(), CBPeripheralDelegateProtocol {
+
+        override fun peripheral(peripheral: CBPeripheral, didDiscoverDescriptorsForCharacteristic: CBCharacteristic, error: NSError?) {
+            info(TAG, "Did Discover Descriptors for Characteristic ${didDiscoverDescriptorsForCharacteristic.UUID.UUIDString}")
```

Comment:
How this logs can be disabled from client code? Or it's temporary while it's in development?

Reply:
I think we should probably remove the logging before merge for now, but I created #301 and #302 so we can keep more around in the future.